src/Npgsql/NpgsqlCommand.cs

```diff
@@ -594,9 +594,7 @@ Task Prepare(bool async)
         {
             var connector = CheckReadyAndGetConnector();
             for (var i = 0; i < Parameters.Count; i++)
-                if (!Parameters[i].IsTypeExplicitlySet)
-                    throw new InvalidOperationException(
-                        "The Prepare method requires all parameters to have an explicitly set type.");
+                Parameters[i].Bind(connector.TypeMapper);
```

Comment:
This change breaks the `RequiresParamTypesSet` test. The code doesn't throw any more, but the test expects an exception. Fix it, please.